### PR TITLE
fix(core,platform,docs): strictTemplates errors fixes

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue(s)

Part of https://github.com/SAP/fundamental-ngx/issues/7564
Part of https://github.com/SAP/fundamental-ngx/issues/7565

## Description

Various fixes for the errors shown after enabling `strictTemplates` & `fullTemplateTypeCheck`. Reasons for fixing described here https://github.com/SAP/fundamental-ngx/issues/7563.

## Screenshots

No visual changes.

### CODE REVIEW

**Comment by N1XUS** [comment]
@platon-rov I see there are failed e2e tests. Could this be due to the changes you've made?

File: apps/docs/src/app/core/component-docs/notification/examples/notification-group/notification-group-example.component.ts [hunk 1]
```diff
@@ -1,4 +1,16 @@
 import { Component } from '@angular/core';
+import { IndicatorStates } from '../../../../../../../../../libs/core/src/lib';
```
Comment (N1XUS) [inline apps/docs/src/app/core/component-docs/notification/examples/notification-group/notification-group-example.component.ts:2]: Wrong path

**Review by InnaAtanasova — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-dialog-example.component.ts [hunk 2]
```diff
@@ -42,6 +42,12 @@ export class WizardDialogExampleComponent {
         });
     }
 
+    // Handle focus on key press
+    /** @hidden */
```
Comment (dmitry-stepanenko) [inline apps/docs/src/app/core/component-docs/wizard/examples/wizard-dialog-example.component.ts:46]: do we need @hidden here? It's docs, not components

File: libs/platform/src/lib/button/button.component.html [hunk 3]
```diff
@@ -14,7 +14,7 @@
     [attr.value]="value"
     [attr.aria-selected]="ariaSelected"
     [attr.aria-disabled]="ariaDisabled"
-    [compact]="contentDensity ? contentDensity !== 'cozy' : ''"
+    [compact]="contentDensity ? contentDensity !== 'cozy' : false"
```
Comment (dmitry-stepanenko) [inline libs/platform/src/lib/button/button.component.html:17]:
```suggestion
    [compact]="contentDensity !== 'cozy'"
```

**Review by dmitry-stepanenko — APPROVED** [review]

**Review by N1XUS — APPROVED** [review]

**Pull request merged**